ios/FluentUI/Bottom Sheet/BottomCommandingController.swift

```diff
@@ -0,0 +1,592 @@
+//
+
+import UIKit
+
+/// Persistent commanding surface displayed at the bottom of the available area.
+///
+/// The presentation style automatically varies depending on the current horizontal `UIUserInterfaceSizeClass`:
+///
+/// `.unspecified` and `.compact` - the surface is displayed as an expandable bottom sheet.
+///
+/// `.regular` -  the surface is displayed as a floating bottom bar.
+///
+/// In both styles, `heroItems` are always presented in a horizontal stack.
+/// Items from the `expandedListSections` are either presented in an expanded sheet or a popover, depending on the current style.
+///
+@objc(MSFBottomCommandingController)
+open class BottomCommandingController: UIViewController {
+
+    /// Items to be displayed in the hero area.
+    @objc open var heroItems: [CommandingItem] = [] {
+        willSet {
+            clearAllItemViews(in: .heroSet)
+        }
+        didSet {
+            if heroItems.count > 5 {
+                assertionFailure("At most 5 hero commands are supported. Only the first 5 items will be used.")
+                heroItems = Array(heroItems.prefix(5))
+            }
+
+            if isHeroCommandStackLoaded {
+                let newViews = heroItems.map { createAndBindHeroCommandView(with: $0) }
+                newViews.forEach { heroCommandStack.addArrangedSubview($0) }
+            }
+        }
+    }
+
+    /// Sections with items to be displayed in the list area.
+    @objc open var expandedListSections: [CommandingSection] = [] {
+        willSet {
+            clearAllItemViews(in: .list)
+        }
+        didSet {
+            expandedListSections.forEach { $0.items.forEach { $0.delegate = self }}
+            if isTableViewLoaded {
+                // Item views and bindings will be lazily created during UITableView cellForRowAt
+                tableView.reloadData()
+            }
+            updateExpandability()
+        }
+    }
+
+    // MARK: - View building and layout
+
+    public override func loadView() {
+        view = BottomSheetPassthroughView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+
+        if traitCollection.horizontalSizeClass == .regular {
+            setupBottomBarLayout()
+        } else {
+            setupBottomSheetLayout()
+        }
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        // On a horizontal size class change the top level sheet / bar surfaces get recreated,
+        // but the item views, containers and bindings persist and are rearranged during the individual setup functions.
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass {
+            bottomSheetController?.willMove(toParent: nil)
+            bottomSheetController?.removeFromParent()
+            bottomSheetController?.view.removeFromSuperview()
+            bottomSheetController = nil
```

Comment:
Did I get too used to it or is it actually more readable with the optionals because of no nesting? It's like in objc when you send a bunch of messages to an object that might be nil but we don't really care.

Reply:
Yeah, I think the main reason is to limit the number of optionals, and thus the number of places code could fail. These all happen to be accessing the same optional, which only makes me want to wrap it inside can `if let` even more.
Definitely a very swift-ism though.